### PR TITLE
You can use non combat abilities when resting as a xeno

### PR DESCRIPTION
**Pull request by BraveMole**
## About The Pull Request

You can use abilities like change type of globule, type of spit, type of pheromone, change leaders, etc... while being rested

Plus, you can rest while being crested as a defender, and while in agi mode as a warrior

## Changelog
:cl:
qol: Almost all non-combat abilities of xenos can be used while resting (changing type of spit, pheros, queen orders, etc etc)
/:cl:

### CODE REVIEW

**Comment by Pariah919** [comment]
first off this is not qol, this is infact balance

**Comment by BraveMole** [comment]
> first off this is not qol, this is infact balance
> 
> this lets hivelords rest and healbeam if i'm reading this right which is infact pretty funny

You can with no issue unrest, beam, then rest back. Since going back to rest is instantanious, this is barely changing anything balance wise

**Comment by InterroLouis** [comment]
First comment is, getting OUT of rest isn't instantaneous, which is what would have delayed you from healing someone who shows up needing it just after you rest. That, however, I've got no issue with. The cooldown on the healbeam is long enough that this doesn't really matter that much, it doesn't let them spam it any more than they would normally. The main issue I have with this is, it allows a Queen to screech while resting, which pushes it straight into balance territory. That plus, there's no way you can really call screech a non combat ability. 
A secondary concern of mine is the resin jelly production. That has no cooldown, and resting, as we know, increases your plasma recovery, that combined with say, recovery pheromones, could lead to a hivelord placing a monumentally huge pile of resin jelly. Or even multiple huge piles of it, or perhaps even hiding resin holes under such large piles of jelly that it has a danger of crashing your game when you right click it, or alt click it, to deal with the hole. This may be possible now, but allowing it while resting would make it a lot quicker and easier to do. 
My third concern here is, actually, the transfer plasma. While not a combat ability, this would allow the user to act as a super plasma battery behind the xeno lines. Essentially, because of the previously mentioned boost to plasma generation, a hivelord could simply...charge up anyone and everyone that steps back from the fight with low health/plasma, however this is not quite as important as the other two because of that requiring it to be, well, resting, in one spot, for an extended period of time. Not quite safe and not quite an interesting way to play. This being an issue or not, I suppose, depends on who is being asked, really, so and on the other hand it does open up a new way to play hivelord, if that's what you're into, so don't take this as me asking for transfer plasma to be excluded. 
Last but not least, I have a few things I would like to suggest get the ability to be used while resting that haven't. First off, word of the Queen and set orders. Changing slashing permissions might as well also get it. 
**Testing has indicated a problem at this point in writing. While you are able to overwatch a xeno as Queen while resting, you can't STOP overwatching them via control middle click until you stand up.**
Just to head this off before it gets said, I did test for being able to spit while resting, and it does not allow that, it just lets you toggle spit types by hitting the button while resting. 
Additionally, I don't see any issue with adding it to things such as runner pounce, hunter pounce, praetorian spray, etc etc, to enable them to select the ability while resting but not use it, as long as it would work the same with those as it does with the spit.

File: code/modules/mob/living/carbon/xenomorph/castes/queen/abilities_queen.dm [hunk 1]
```diff
@@ -122,6 +122,7 @@
 	cooldown_timer = 100 SECONDS
 	keybind_flags = XACT_KEYBIND_USE_ABILITY
 	keybind_signal = COMSIG_XENOABILITY_SCREECH
+	use_state_flags = XACT_USE_LYING
```
Comment (SplinterGP) [inline code/modules/mob/living/carbon/xenomorph/castes/queen/abilities_queen.dm:125]: Uh, Why ?

**Review by SplinterGP — CHANGES_REQUESTED** [review]

**Comment by SplinterGP** [comment]
i think keybinds kinda bypass some of those skills somehow ? just dont remember how

File: code/modules/mob/living/carbon/xenomorph/castes/queen/abilities_queen.dm [hunk 2]
```diff
@@ -244,10 +245,11 @@
 /datum/action/xeno_action/watch_xeno
 	name = "Watch Xenomorph"
 	action_icon_state = "watch_xeno"
-	mechanics_text = "See from the target Xenomorphs vision."
+	mechanics_text = "See from the target Xenomorphs vision. Cntrl middle click to stop observing"
```
[the remark this answers is not in the document]
Comment (BraveMole) [reply code/modules/mob/living/carbon/xenomorph/castes/queen/abilities_queen.dm:248]: Apparently did't work, strange

**Comment by BraveMole** [comment]
> First comment is, getting OUT of rest isn't instantaneous, which is what would have delayed you from healing someone who shows up needing it just after you rest. That, however, I've got no issue with. The cooldown on the healbeam is long enough that this doesn't really matter that much, it doesn't let them spam it any more than they would normally. The main issue I have with this is, it allows a Queen to screech while resting, which pushes it straight into balance territory. That plus, there's no way you can really call screech a non combat ability.
> A secondary concern of mine is the resin jelly production. That has no cooldown, and resting, as we know, increases your plasma recovery, that combined with say, recovery pheromones, could lead to a hivelord placing a monumentally huge pile of resin jelly. Or even multiple huge piles of it, or perhaps even hiding resin holes under such large piles of jelly that it has a danger of crashing your game when you right click it, or alt click it, to deal with the hole. This may be possible now, but allowing it while resting would make it a lot quicker and easier to do.
> My third concern here is, actually, the transfer plasma. While not a combat ability, this would allow the user to act as a super plasma battery behind the xeno lines. Essentially, because of the previously mentioned boost to plasma generation, a hivelord could simply...charge up anyone and everyone that steps back from the fight with low health/plasma, however this is not quite as important as the other two because of that requiring it to be, well, resting, in one spot, for an extended period of time. Not quite safe and not quite an interesting way to play. This being an issue or not, I suppose, depends on who is being asked, really, so and on the other hand it does open up a new way to play hivelord, if that's what you're into, so don't take this as me asking for transfer plasma to be excluded.
> Last but not least, I have a few things I would like to suggest get the ability to be used while resting that haven't. First off, word of the Queen and set orders. Changing slashing permissions might as well also get it.
> **Testing has indicated a problem at this point in writing. While you are able to overwatch a xeno as Queen while resting, you can't STOP overwatching them via control middle click until you stand up.**
> Just to head this off before it gets said, I did test for being able to spit while resting, and it does not allow that, it just lets you toggle spit types by hitting the button while resting.
> Additionally, I don't see any issue with adding it to things such as runner pounce, hunter pounce, praetorian spray, etc etc, to enable them to select the ability while resting but not use it, as long as it would work the same with those as it does with the spit.

Yeah, good remarks.

So I'm gonna remove jelly and screech. Don't think med beam and transfer plasma are such balance issues, compared to the convenience brought.

Overwatch for the queen is apparently bugged, you're supposed to be able to use ctrl mdl click to leave overwatch. You can though leave overwatch by clicking the overwatch button again and clicking cancel, but i'll make a new way to leave it that's more convenient

For the other abilities (word of the queen), it's a verb, not an ability, but it's probably possible to make them usuable when resting, and a good idea

File: code/modules/mob/living/carbon/xenomorph/abilities.dm [hunk 3]
```diff
@@ -448,6 +449,7 @@
 	var/transfer_delay = 2 SECONDS
 	var/max_range = 2
 	keybind_signal = COMSIG_XENOABILITY_TRANSFER_PLASMA
+	use_state_flags = XACT_USE_LYING
```
Comment (TiviPlus) [inline code/modules/mob/living/carbon/xenomorph/abilities.dm:452]: No

File: code/modules/mob/living/carbon/xenomorph/abilities.dm [hunk 4]
```diff
@@ -781,6 +783,7 @@
 	mechanics_text = "Spit neurotoxin or acid at your target up to 7 tiles away."
 	ability_name = "xeno spit"
 	keybind_signal = COMSIG_XENOABILITY_XENO_SPIT
+	use_state_flags = XACT_USE_LYING
```
Comment (TiviPlus) [inline code/modules/mob/living/carbon/xenomorph/abilities.dm:786]: No

File: code/modules/mob/living/carbon/xenomorph/castes/hivelord/abilities_hivelord.dm [hunk 5]
```diff
@@ -207,6 +208,7 @@ GLOBAL_LIST_INIT(thickenable_resin, typecacheof(list(
 	plasma_transfer_amount = PLASMA_TRANSFER_AMOUNT * 4
 	transfer_delay = 0.5 SECONDS
 	max_range = 7
+	use_state_flags = XACT_USE_LYING
```
Comment (TiviPlus) [inline code/modules/mob/living/carbon/xenomorph/castes/hivelord/abilities_hivelord.dm:211]: No

File: code/modules/mob/living/carbon/xenomorph/castes/hivelord/abilities_hivelord.dm [hunk 6]
```diff
@@ -281,6 +283,7 @@ GLOBAL_LIST_INIT(thickenable_resin, typecacheof(list(
 	cooldown_timer = 12.5 SECONDS
 	plasma_cost = 200
 	keybind_signal = COMSIG_XENOABILITY_HEALING_INFUSION
+	use_state_flags = XACT_USE_LYING
```
Comment (TiviPlus) [inline code/modules/mob/living/carbon/xenomorph/castes/hivelord/abilities_hivelord.dm:286]: No

**Review by TiviPlus — CHANGES_REQUESTED** [review]

**Comment by SpaceLoveSs13** [comment]
Only things I have concerns about other than those previously mentioned and answered, are a) Are we sure Defenders won't get any extra advantage with crest and rest? And b) Being on agile mode while resting means warrior can very quickly run away even if they are resting instead of being penalized for resting in dangerous zone. 

If my concern about B does however get approved, will hivelord also have this ability with their resin walker?

**Comment by BraveMole** [comment]
> If my concern about B does however get approved, will hivelord also have this ability with their resin walker?

hivelord can rest with resin walker AFAIK

File: code/modules/mob/living/carbon/xenomorph/abilities.dm (hunk 4 above)
Comment (BraveMole) [reply code/modules/mob/living/carbon/xenomorph/abilities.dm:786]: This doesn't allow to spit. This just allow to change spit, cause the buttons are merged

Comment (InterroLouis) [reply code/modules/mob/living/carbon/xenomorph/abilities.dm:785]: Was my initial comment too long? I did already say that it didn't allow you to spit while resting, just to choose your spit type.

**Comment by SplinterGP** [comment]
> Only things I have concerns about other than those previously mentioned and answered, are a) Are we sure Defenders won't get any extra advantage with crest and rest? And b) Being on agile mode while resting means warrior can very quickly run away even if they are resting instead of being penalized for resting in dangerous zone. 
> 
> If my concern about B does however get approved, will hivelord also have this ability with their resin walker?

If they rest in agility i think the armor debuff/buff stays so its a bit killing yourself

**Comment by SplinterGP** [comment]
Also @BraveMole using rest keybind fully bypasses any skill and allows you to rest no matter what(which also is dangerous if you use a skill with channeling, because then you cant get up again) id recommend fixing it, also beware putting any skill with channel to be used while resting as well

**Comment by BraveMole** [comment]
> Also @BraveMole using rest keybind fully bypasses any skill and allows you to rest no matter what(which also is dangerous if you use a skill with channeling, because then you cant get up again) id recommend fixing it, also beware putting any skill with channel to be used while resting as well

@SplinterGP I still don't understand what you want. I'm not fixing the rest keybind here. 
No skill with channel can be used while resting, appart from headbite and cocoon, but that's far beyond the scope of this PR
It's a simple PR, no need to add things to it

**Comment by BraveMole** [comment]
> Also @BraveMole using rest keybind fully bypasses any skill and allows you to rest no matter what(which also is dangerous if you use a skill with channeling, because then you cant get up again) id recommend fixing it, also beware putting any skill with channel to be used while resting as well

Not relevant to this PR. You want to fix, you do it. And you certainly don't call me lazy for not doing it

**Review by SplinterGP — APPROVED** [review]

**Review by TiviPlus — APPROVED** [review]

**Pull request merged**